static/js/components/channels/ChannelCreateDialog.js

```diff
@@ -0,0 +1,145 @@
+// @flow
+/* global SETTINGS: false */
+import Dialog from "material-ui/Dialog"
+import React from "react"
+import TextField from "material-ui/TextField"
+import Grid, { Cell } from "react-mdl/lib/Grid"
+import R from "ramda"
+
+import { dialogActions } from "../inputs/util"
+import { renderFilterOptions } from "../email/lib"
+import { FETCH_PROCESSING } from "../../actions"
+
+import type { ChannelState, Filter } from "../../flow/discussionTypes"
+import type { AvailableProgram } from "../../flow/enrollmentTypes"
+
+type ChannelCreateDialogProps = {
+  channelDialog: ChannelState,
+  dialogVisibility: boolean,
+  currentProgramEnrollment: ?AvailableProgram,
+  closeAndClearDialog: () => void,
+  closeAndCreateDialog: () => void,
+  updateEmailFieldEdit: () => void
+}
+
+export default class ChannelCreateDialog extends React.Component {
+  props: ChannelCreateDialogProps
+
+  closeAndClear = (): void => {
+    const { closeAndClearDialog } = this.props
+    closeAndClearDialog()
+  }
+
+  closeAndCreate = (): void => {
+    const { closeAndCreateDialog } = this.props
+    closeAndCreateDialog()
+  }
+
+  showValidationError = (fieldName: string): ?React$Element<*> => {
+    const {
+      channelDialog: { validationErrors, validationVisibility }
+    } = this.props
+    const isVisible = R.propOr(false, fieldName)
+    const val = validationErrors[fieldName]
+    if (isVisible(validationVisibility) && val !== undefined) {
+      return <span className="validation-error">{val}</span>
+    }
+  }
+
+  renderChannelFilters(program: AvailableProgram, filters: ?Array<Filter>) {
+    if (!filters || R.isEmpty(filters)) {
+      return (
+        <div className="sk-selected-filters-list">
+          <div className="sk-selected-filters-option sk-selected-filters__item">
+            <div className="sk-selected-filters-option__name">
+              All users in {program.title}
+            </div>
+          </div>
+        </div>
+      )
+    }
+    return (
+      <div className="sk-selected-filters-list">
+        {renderFilterOptions(filters)}
+      </div>
+    )
+  }
+
+  render() {
+    const {
+      channelDialog: { inputs, filters, validationErrors, fetchStatus },
+      dialogVisibility,
+      currentProgramEnrollment,
+      updateEmailFieldEdit
+    } = this.props
+
+    if (!currentProgramEnrollment) {
+      return null
+    }
+
+    return (
+      <Dialog
+        title="Create New Channel"
+        titleClassName="dialog-title"
+        contentClassName="dialog create-channel-dialog"
+        className="create-channel-dialog-wrapper"
+        open={dialogVisibility}
+        onRequestClose={this.closeAndClear}
+        actions={dialogActions(
+          this.closeAndClear,
+          this.closeAndCreate,
+          fetchStatus === FETCH_PROCESSING,
+          "Create",
+          "",
+          !R.isEmpty(validationErrors)
```

Comment:
I think you need to include `fetchStatus === FETCH_PROCESSING` in this condition

Reply:
Actually, it's fine this way, because `SpinnerButton` considers itself `disabled` if it's spinning: https://github.com/mitodl/micromasters/blob/master/static/js/components/SpinnerButton.js#L36